rfcs/react-components/styles-handbook.md

```diff
@@ -0,0 +1,771 @@
+
+```jsx
+import { makeStyles } from '@griffel/react';
+
+const useClasses = makeStyles({
+  test: {
+    color: 'orange',
+
+    '> *': {
+      color: 'red',
+    },
+    '> h1': {
+      color: 'magenta',
+    },
+    '> div': {
+      color: 'green',
+    },
+    '> .ui-button': {
+      color: 'blue',
+    },
+  },
+});
+
+function App() {
+  const classes = useClasses();
+
+  return (
+    <>
+      <div className={classes.test}>
+        <h1>Hello World</h1>
+        <button class="ui-button">A button</button>
+      </div>
+
+      {/* Renders 500 empty divs */}
+      {Array.from({ length: 500 }, (_, i) => (
+        <div key={i} />
+      ))}
+    </>
+  );
+}
+```
+
+#### Measurements
+
+##### No selector (best)
+
+Matches only an element where is applied.
+
+- selector `.fe3e8s9`
+- match_attempts **1**
+- match_count **1**
+
+##### `> *` (worst)
+
+"\*" matches any element on a page.
+
+- selector `.fzbuleu > *`
+- match_attempts **502**
+- match_count **1**
+
+##### `> h1` & `> div` (non ideal)
+
+Targets all `h1` & `div` tags.
+
+- selector `.fohk16y > h1`
+- match_attempts **1**
+- match_count **1**
+
+In this example performs better as page has a single `h1` tag, this will not be a case in the real app that it is visible on the next example with `div`.
+
+- selector `.fq4d7o6 > div`
+- match_attempts **501**
+- match_count **1**
+
+#### Targeting a classname (better)
+
+The most effective out there as we target only elements with a specific classname. However, it will not save from the case with 1k buttons in the app.
+
+- selector `.fqhvij7 > .ui-button`
+- match_attempts **1**
+- match_count **1**
+
+# Best practices
+
+## Writting styles
+
+### Use `tokens` over direct colors
+
+Fluent UI React v9 provides [design tokens][fluent-colors] to provide consistent way of theming via `tokens` provided by `@fluentui/react-theme`.
+
+```js
+import { makeStyles } from '@griffel/react';
+import { tokens } from '@fluentui/react-theme';
+
+const useClasses = makeStyles({
+  // ❌ Don't do
+  rootA: { color: 'red' /* brand foreground */ },
+  // ✅ Do
+  rootB: { color: tokens.colorBrandForeground1 },
+});
+```
+
+Exact colors should be never used in Fluent UI code and are not recommended by be used in applications. The exception are [system-colors][mdn-system-colors] used by forced colors mode:
+
+```js
+import { makeStyles } from '@griffel/react';
+
+const useClasses = makeStyles({
+  button: { color: 'ButtonText' },
+});
+```
+
+### Avoid duplication in rule definitions
+
+Styles written for components should follow these rules:
+
+- base styles should contain the most of CSS definitions that are applicable to base state
+- styles for permutations should be granular and don't contain definitions that are present in base styles
+
+```js
+import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
+import { tokens } from '@fluentui/react-theme';
+
+const useClasses = makeStyles({
+  base: {
```

Comment:
should we be consistently using `makeResetStyles` for these examples for base styles?

Reply:
We should, done in 9e9d2e4